### PR TITLE
ADEN-12667 Fix GPT errors on TV Guide

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

**Review by dmnsobczak — COMMENTED** [review]
Let's discuss, my idea how to handle it without changes in shared services 🤔 
https://github.com/Wikia/ad-engine/pull/1757/files

File: src/core/services/slot-injector.ts [hunk 1]
```diff
@@ -64,6 +64,13 @@ class SlotInjector {
 			slotConfig.avoidConflictWith = [config.avoidConflictWith];
 		}
 
+		if (
+			slotConfig.insertMethod === 'none' &&
+			this.getDisablePushOnScroll(disablePushOnScroll, config) === false
+		) {
+			context.push('state.adStack', { id: slotName });
```
[the remark this answers is not in the document]
Comment (nandy-andy) [reply src/core/services/slot-injector.ts:71]: Yeah but it's been broken before 😊 https://github.com/Wikia/ad-engine/pull/1755/files#diff-ac3e1716bb48c14ffcf4bc3f10402864d1768e12a83eeab21fd6f2d187516080R83

File: src/platforms/news-and-ratings/tvguide/setup/context/slots/tvguide-slots-context.setup.ts [hunk 2]
```diff
@@ -218,16 +218,36 @@ export class TvGuideSlotsContextSetup implements DiProcess {
...
 				],
-				incremental: true,
 				targeting: {
 					pos_nr: 'inc',
+					loc: 'middle',
+					pos: 'incontent-leader-plus-inc',
+					rv: 1,
+				},
+				parentContainerSelector: 'none', // server-side rendered container
+				insertIntoParentContainerMethod: 'none', // server-side rendered container
```
[the remark this answers is not in the document]
Comment (nandy-andy) [reply src/platforms/news-and-ratings/tvguide/setup/context/slots/tvguide-slots-context.setup.ts:239]: Discussed offline.

File: src/platforms/news-and-ratings/tvguide/setup/dynamic-slots/tvguide-dynamic-slots.setup.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { context, DiProcess, utils } from '@wikia/ad-engine';
 
 export class TvGuideDynamicSlotsSetup implements DiProcess {
+	private alreadyPushedSlots = {};
```
[the remark this answers is not in the document]
Comment (nandy-andy) [reply src/platforms/news-and-ratings/tvguide/setup/dynamic-slots/tvguide-dynamic-slots.setup.ts:4]: Discussed offline that it's more complicated ;-)

File: src/core/services/slot-creator.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import {
 	logger,
 } from '../utils';
 import { context } from './context-service';
-export type insertMethodType = 'append' | 'prepend' | 'after' | 'before';
+export type insertMethodType = 'append' | 'prepend' | 'after' | 'before' | 'none';
```
[the remark this answers is not in the document]
Comment (nandy-andy) [reply src/core/services/slot-creator.ts:12]: Discussed offline.

**Review by dmnsobczak — APPROVED** [review]

**Review by dmnsobczak — APPROVED** [review]

**Pull request merged**